### DISCUSSION TITLE
About sanboot module - use iso file to install Debian

### DISCUSSION THREAD

**Discussion by mabps** [ipxe/ipxe#644] [Q&A]
Hi,
on your forum it is written that to run from an ISO file, it is enough to do this:
https://forum.ipxe.org/showthread.php?tid=3728

```
# Start Windows 7 installer DVD
sanboot --no-describe --drive 0x81 ${boot-url}/windows-7/ultimate-x64.iso || goto failed
goto start

```

by analogy, I'm trying to run a Debian installation CD

```
:debian-11.3.0-i386-DVD-1.iso
set keep-san 1
set syslog 192.168.0.1
sanboot --no-describe -k --drive 0x81 http://192.168.0.1/iso/debian-11.3.0-i386-DVD-1.iso
goto start

```
when you run this menu item - the disk starts, but during installation - Debian cannot find the media
I thought that Sanboot creates an emulation of the installation disk and the Debian installer gets access to it through the sanboot "driver".
Where did I go wrong, what am I doing wrong?

(I'm using the original debian ISOs? like this https://cdimage.debian.org/debian-cd/current/i386/iso-cd/debian-11.3.0-i386-netinst.iso )
(After starting from ISO by iPXE- the network interface is not available in the installation system, it is not possible to connect a network repository)

**Comment by NiKiZe** [comment]
If you use iscsi, and the os has support and NIC drivers, then you can boot Isos this way.

However booting a iso over http is often not possible. Here is one of several explanations https://github.com/ipxe/ipxe/discussions/476#discussioncomment-1371552

**Reply by mabps** [reply]
thank you for you answer
i will think about it

**Comment by stappersg** [comment]
> when you run this menu item - the disk starts, but during installation - Debian cannot find the media

Short `root=/dev/something`.

Long:
`sanboot` makes booting from networked storage possible.  The _boot_ brings the Linux kernel alive. Then the kernel needs be aware where to find the _root device_. That's something the _bootloader_ usually tells.  I don't know how to add `root=/dev/something` to `sanboot`, neither the correct name for _something_.

**Comment by stappersg** [comment]
@Arkadiy-SPb congratulations on `644`,  a.k.a. `-rwxr-xr-x`,   `:-)`

I hope you can appreciate https://wiki.debian.org/DebianInstaller/NetbootAssistant

**Reply by mabps** [reply]
> @Arkadiy-SPb congratulations on `644`, a.k.a. `-rwxr-xr-x`, `:-)`
> 
> I hope you can appreciate https://wiki.debian.org/DebianInstaller/NetbootAssistant

thanks for the answer and explanation
I haven't seen this article before, it looks interesting, thanks for the link

now I try to make universal PXE boot server with use Ansible (for make installation easier), maybe it's will interest for you, I will be glad of your comments
https://github.com/Arkadiy-SPb/PXE-server-ansible-playbook

**Reply by stappersg** [reply]
Oops, off by `111`,  `644` is `-rw-r--r--`, looking forward to `755`.
